src/sage/groups/perm_gps/permgroup.py

```diff
@@ -2118,8 +2193,32 @@ def _repr_(self):
             'Permutation Group with generators [(2,3,4), (1,2,3)]'
             sage: AlternatingGroup(4)._repr_()
             'Alternating group of order 4!/2 as a permutation group'
+
+        The domain is printed if it is not of the form `{1,...,n}`,
+        where `n` is the largest moved point::
+
+            sage: PermutationGroup([(1,3)], domain=[1,3])
+            Permutation Group with generators [(1,3)] and domain {1, 3}
+
+            sage: PermutationGroup([(1,3)], domain=[1,2,3])
+            Permutation Group with generators [(1,3)]
+
+            sage: PermutationGroup([(1,3)], domain=[1,3,2])
+            Permutation Group with generators [(1,3)] and domain {1, 3, 2}
+
+            sage: PermutationGroup([(1,3)], domain=[1,2,3,4])
+            Permutation Group with generators [(1,3)] and domain {1, 2, 3, 4}
+
+        TESTS::
+
+            sage: PermutationGroup([], domain=[])
+            Permutation Group with generators [()]
+
         """
-        return "Permutation Group with generators %s" % list(self.gens())
+        if (self._has_natural_domain()
+            and self.largest_moved_point() == max(self.domain(), default=0)):
```

Comment:
```suggestion
                and self.largest_moved_point() == max(self.domain(), default=0)):
```
(need to indent because of the following line)